core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/StackV4RequestToStackConverter.java

```diff
@@ -180,15 +182,8 @@ private void updateCustomDomainOrKerberos(StackV4Request source, Stack stack) {
             stack.setClusterNameAsSubdomain(source.getCustomDomain().isClusterNameAsSubdomain());
             stack.setHostgroupNameAsHostname(source.getCustomDomain().isHostgroupNameAsHostname());
         } else if (!isTemplate(source)) {
-            Optional<KerberosConfig> kerberosConfig = kerberosConfigService.get(stack.getEnvironmentCrn(), stack.getName());
-            kerberosConfig.ifPresent(kb -> {
-                if (kb.getType() == KerberosType.ACTIVE_DIRECTORY) {
-                    if (isEmpty(kb.getRealm())) {
-                        throw new BadRequestException("Realm cannot be null in case of ACTIVE_DIRECTORY");
-                    }
-                    stack.setCustomDomain(kb.getRealm().toLowerCase());
-                }
-            });
+            String domain = freeIPAAdapter.getDomain(stack.getEnvironmentCrn());
+            stack.setCustomDomain(domain);
```

Comment:
I think the original code set the customDomain only in case of ACTIVE_DIRECTORY, since AD is not supported I think the entire else if part can be deleted